sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Infrastructure/StorageTestEnvironment.cs

```diff
@@ -15,42 +15,74 @@ namespace Azure.Messaging.EventHubs.Processor.Tests
+        public static bool WasStorageAccountCreated => ActiveStorageAccount.IsValueCreated && ActiveStorageAccount.Value.WasStorageAccountCreated;
 
         /// <summary>
         ///   The name of the Azure storage account to be used for Live tests.
         /// </summary>
         ///
         /// <value>The name will be determined by creating an ephemeral Azure storage account for the test execution.</value>
         ///
-        public static string StorageAccountName => s_activeStorageAccount.Value.Name;
+        public static string StorageAccountName => ActiveStorageAccount.Value.Name;
 
         /// <summary>
         ///   The connection string for the Azure storage instance to be used for Live tests.
         /// </summary>
         ///
         /// <value>The connection string will be determined by creating an ephemeral Azure storage account for the test execution.</value>
         ///
-        public static string StorageConnectionString => s_activeStorageAccount.Value.ConnectionString;
+        public static string StorageConnectionString => ActiveStorageAccount.Value.ConnectionString;
+
+        /// <summary>
+        ///   It tries to read the <see cref="ActiveStorageAccountConnectionString" />.
+        ///   If not found, it creates a new storage account on Azure.
+        /// </summary>
+        ///
+        /// <returns>The active Azure storage account for this test run.</returns>
+        ///
+        private static StorageScope.StorageProperties EnsureStorageAccount()
+        {
+            if (!string.IsNullOrEmpty(ActiveStorageAccountConnectionString.Value))
+            {
+                return StorageScope.PopulateStoragePropertiesFromConnectionString(ActiveStorageAccountConnectionString.Value);
+            }
+
+            return BuildStoragePropertiesWithNewStorageAccount();
+        }
+
+        /// <summary>
+        ///   Reads an environment variable.
+        /// </summary>
+        ///
+        /// <param name="variableName">The name of the environment variable to read.</param>
+        ///
+        /// <returns>The value of the environment variable, if present and populated; null otherwise</returns>
+        ///
+        private static string ReadEnvironmentVariable(string variableName) =>
+            Environment.GetEnvironmentVariable(variableName);
 
         /// <summary>
         ///   Requests creation of an Azure storage account to use for a specific test run,
         ///   transforming the asynchronous request into a synchronous one that can be used with
         ///   lazy instantiation.
         /// </summary>
         ///
-        /// <returns>The active Azure storage account for this test run./returns>
+        /// <returns>The active Azure storage account for this test run.</returns>
         ///
-        private static StorageScope.StorageProperties CreateStorageAccount() =>
+        private static StorageScope.StorageProperties BuildStoragePropertiesWithNewStorageAccount() =>
```

Comment:
I think we should definitely keep the word `Create` or similar here to be clear that we're creating a new Azure resource.   The `NewStorageAccount` kind of does that, but I don' t think it is as explicit.   

Personally, I think the original name would be my preference.